processor/datadogprocessor/agent.go

```diff
@@ -142,6 +146,9 @@ func (p *traceagent) Ingest(ctx context.Context, traces ptrace.Traces) {
 	rspanss := traces.ResourceSpans()
 	for i := 0; i < rspanss.Len(); i++ {
 		rspans := rspanss.At(i)
+		// Stats will be computed for p. Mark the original resource spans to ensure that they don't
+		// get computed twice in case these spans pass through here again.
+		rspans.Resource().Attributes().PutBool(keyStatsComputed, true)
```

Comment:
This needs to go _after_ the call below. The call below [checks for it](https://github.com/DataDog/datadog-agent/blob/8228f5bff0a7a2d2e91db002ad4d6a531df4c303/pkg/trace/api/otlp.go#L307). If you put it here, it will avoid computing stats. The first call has to compute them. The second one not. I am _really_ hoping that the unit tests catch this mistake :)

Reply:
Fixed, PTAL